tokio-tcp/src/stream.rs

```diff
@@ -131,6 +188,32 @@ impl TcpStream {
     ///
     /// * `ready` includes writable.
     /// * called from outside of a task context.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # extern crate mio;
+    /// # extern crate tokio_io;
+    /// # extern crate tokio_tcp;
+    /// # extern crate futures;
+    /// # use futures::Async;
+    /// # use futures::Future;
+    /// # use tokio_tcp::TcpStream;
+    /// # use std::net::SocketAddr;
```

Comment:
Looks like there's some inconsistency here where we aren't showing any of the use statements at all?

Reply:
So I wasn't sure exactly what to show and what to hide. Do you have a preference or **rule of thumb** I could follow?

Reply:
Another thing to consider is that the `TcpListener` and `TcpStream` are mostly not used in isolation, which makes it a bit hard to provide meaningful examples.

For instance, the `tinyhttp.rs` in **Examples** has this snippet:
```rust
    let addr = env::args().nth(1).unwrap_or("127.0.0.1:8080".to_string());
    let addr = addr.parse::<SocketAddr>()?;

    let listener = TcpListener::bind(&addr)?;
    println!("Listening on: {}", addr);

    tokio::run({
        listener.incoming()
            .map_err(|e| println!("failed to accept socket; error = {:?}", e))
            .for_each(|socket| {
                process(socket);
                Ok(())
            })
    });
```
This actually shows how to use the TcpListener with the Tokio runtime and could be really useful to have in the documentation but this is hard - the doctests will fail since the runtime is not part of the tcp crate.